lib/reline/unicode.rb

```diff
@@ -54,51 +54,26 @@ def self.escape_for_print(str)
 
   require 'reline/unicode/east_asian_width'
 
-  HalfwidthDakutenHandakuten = /[\u{FF9E}\u{FF9F}]/
-
-  MBCharWidthRE = /
-    (?<width_2_1>
-      [#{ EscapedChars.map {|c| "\\x%02x" % c.ord }.join }] (?# ^ + char, such as ^M, ^H, ^[, ...)
-    )
-  | (?<width_3>^\u{2E3B}) (?# THREE-EM DASH)
-  | (?<width_0>^\p{M})
-  | (?<width_2_2>
-      #{ EastAsianWidth::TYPE_F }
-    | #{ EastAsianWidth::TYPE_W }
-    )
-  | (?<width_1>
-      #{ EastAsianWidth::TYPE_H }
-    | #{ EastAsianWidth::TYPE_NA }
-    | #{ EastAsianWidth::TYPE_N }
-    )(?!#{ HalfwidthDakutenHandakuten })
-  | (?<width_2_3>
-      (?: #{ EastAsianWidth::TYPE_H }
-        | #{ EastAsianWidth::TYPE_NA }
-        | #{ EastAsianWidth::TYPE_N })
-      #{ HalfwidthDakutenHandakuten }
-    )
-  | (?<ambiguous_width>
-      #{EastAsianWidth::TYPE_A}
-    )
-  /x
-
   def self.get_mbchar_width(mbchar)
     ord = mbchar.ord
-    if (0x00 <= ord and ord <= 0x1F) # in EscapedPairs
+    if ord <= 0x1F # in EscapedPairs
       return 2
-    elsif (0x20 <= ord and ord <= 0x7E) # printable ASCII chars
+    elsif ord <= 0x7E # printable ASCII chars
       return 1
     end
-    m = mbchar.encode(Encoding::UTF_8).match(MBCharWidthRE)
-    case
-    when m.nil? then 1 # TODO should be U+FFFD � REPLACEMENT CHARACTER
-    when m[:width_2_1], m[:width_2_2], m[:width_2_3] then 2
-    when m[:width_3] then 3
-    when m[:width_0] then 0
-    when m[:width_1] then 1
-    when m[:ambiguous_width] then Reline.ambiguous_width
+    utf8_mbchar = mbchar.encode(Encoding::UTF_8)
+    ord = utf8_mbchar.ord
+    chunk_index = Reline::Unicode::EastAsianWidth::CHUNK_LAST.bsearch_index { |o| ord <= o }
+    size = Reline::Unicode::EastAsianWidth::CHUNK_WIDTH[chunk_index]
```

Comment:
The `chunk_index` can be `nil` and then line 67 will fail!

```suggestion
    chunk_index = Reline::Unicode::EastAsianWidth::CHUNK_LAST.bsearch_index { |o| ord <= o }
    return 1 unless chunk_index
    size = Reline::Unicode::EastAsianWidth::CHUNK_WIDTH[chunk_index]
```

Reply:
Thank you for the feedback. Can you provide an input character that chunk_index will be nil?

I think `utf8_mbchar.ord` is always less than or equal to `0x10ffff` (maximum codepoint of unicode)

The last two values of CHUNK_LAST are: `0x10fffd` and `0x7fffffff`.
The max value of 32bit signed integer `0x7fffffff` is added at the end of CHUNK_LAST to prevent bsearch_index to return nil.

Reply:
Oh - I missed the additional integer at the end!

I only saw the comment about how the table was generated and missed the "non-unicode" addition…

Thanks for clarification!